### PR TITLE
feat: rapidproto generator support for gogo unmarshalling

### PR DESCRIPTION
**Pull request by kocubinski**
These changes to the generator options are required for the testing suite introduced for the amino json encoder in https://github.com/cosmos/cosmos-sdk/pull/14877 which heavily leverages the rapidproto package.

### CODE REVIEW

File: rapidproto/rapidproto.go [hunk 1]
```diff
@@ -15,20 +17,70 @@ func MessageGenerator[T proto.Message](x T, options GeneratorOptions) *rapid.Gen
...
+	NoEmptyLists bool
+
+	// DisallowNilMessages will cause the generator to not generate nil messages to protoreflect.MessageKind fields
+	DisallowNilMessages bool
+
+	// GogoUnmarshalCompatibleDecimal will cause the generator to generate decimal values which, when serialized as
+	// either bytes or a string, are compatible with gogo's unmarshaler.  These fields are identified by the presence
+	// of (amino.encoding = "cosmos_dec_bytes") or (cosmos_proto.scalar = "cosmos.Dec") annotations.
+	// If this option is set, the AminoEncodingExtension must also be set.
+	GogoUnmarshalCompatibleDecimal bool
```
Comment (aaronc) [inline rapidproto/rapidproto.go:43]: Can't we just have some way of registering special handlers for amino.encoding and cosmos_proto.scalar with the generator?

Comment (kocubinski) [reply rapidproto/rapidproto.go:43]: That would work too, although I'm not sure it's any simpler.

Comment (kocubinski) [reply rapidproto/rapidproto.go:43]: On the other hand, it would remove this horribly specific field name.

Comment (amaurym) [reply rapidproto/rapidproto.go]: I also prefer this new FieldMapper, it removes the dep on SDK concepts such as aminoExtension and cosmos.Dec

**Review by amaurym — APPROVED** [review]

File: rapidproto/rapidproto.go [hunk 2]
```diff
@@ -15,20 +17,60 @@ func MessageGenerator[T proto.Message](x T, options GeneratorOptions) *rapid.Gen
...
+
 type GeneratorOptions struct {
-	AnyTypeURLs []string
-	Resolver    protoregistry.MessageTypeResolver
+	AnyTypeURLs    []string
+	InterfaceHints map[string]string
+	Resolver       protoregistry.MessageTypeResolver
+
+	// NoEmptyLists will cause the generator to not generate empty lists
+	// Recall that an empty list will marshal (and unmarshal) to null.   Some encodings may treat these states
```
Comment (samricotta) [inline rapidproto/rapidproto.go:37]:
```suggestion
	// Recall that an empty list will marshal (and unmarshal) to null. Some encodings may treat these states
```
just had some extra spacing before `Some`

**Pull request merged**